pkg/apis/infra/v1alpha1/cluster_types.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright Kurator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// Cluster is the schema for the clusters API
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type Cluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ClusterSpec   `json:"spec,omitempty"`
+	Status            ClusterStatus `json:"status,omitempty"`
+}
+
+// ClusterSpect defines the desired state of Cluster
+type ClusterSpec struct {
+	// Provider is the provider of the cluster.
+	Provider ClusterProviderType `json:"provider"`
```

Comment:
**Provider** would be more appropriate as the provider of the Kubernetes distribution.

We also need a field to indicate aws, gcp, aliyun, and so on. For example，
// Type represents the type of the Infrastructure.
Type *string `json:"type"`